tests/stress_tests/scripts/run_memcheck.py

```diff
@@ -158,7 +176,20 @@ def main():
         # create timeline report
         if args.timeline_report:
             create_memcheck_report(records, args.db_url, args.db_collection, args.timeline_report)
-            logging.info('Created memcheck report %s', args.timeline_report)
+            logging.info('Created memcheck timeline report %s', args.timeline_report)
+
+        # compare runs and prepare report
+        if args.compare:
+            if os.path.isdir(args.compare):
+                references = get_memcheck_records(source=args.compare)
+            else:
+                query = dict(item.split("=") for item in args.compare.split(","))
+                references = get_db_memcheck_records(query=query,
+                                                     db_collection=args.ref_db_collection,
+                                                     db_name=DATABASE, db_url=args.db_url)
+            compare_memcheck_2_runs(cur_values=records, references=references, output_file=args.comparison_report)
```

Comment:
The comparison should return error and errors should be printed into stderr. A comparison report html should be optional.

Reply:
Done. Please, review once again

Reply:
Comparison report is already optional. `compare_memcheck_2_runs` dumps file only if `output_file` means True (not None).